### PR TITLE
Kafka To Pub/Sub Template Blogpost

### PR DESCRIPTION
**Pull request by KhaninArtur**
Blog post with an announcement of the first Beam template.
* [The link to PR with the template](https://github.com/apache/beam/pull/13112)
* [The link to JIRA issue](https://issues.apache.org/jira/browse/BEAM-11065)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: website/www/site/content/en/blog/kafka-to-pubsub-template.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+There are two ways to execute the pipeline.
+1. Locally. This way has many options - run directly from your IntelliJ, or create `.jar` file and run
+it in the terminal, or use your favourite method.
+2. In [Google Cloud](https://cloud.google.com/). Using Google Cloud [Dataflow](https://cloud.google.com/dataflow) with `gcloud`
+command-line tool you can create a Flex Template out of this Beam template and execute it in Google Cloud Platform.
+
+The only difference between local and cloud execution is that **Flex version of this template doesn't support SSL configuration.**
+
+# Next Steps
+Give this **first Beam template** a try, and let us know your feedback adding comments to [BEAM-11065 JIRA ticket](https://issues.apache.org/jira/browse/BEAM-11065) or sending an email in [this thread](https://lists.apache.org/thread.html/r8db90b3a1271188afb31ea52ccfed231f276f26ded3855d90084fe85%40%3Cdev.beam.apache.org%3E)!
```
Comment (aaltay) [inline website/www/site/content/en/blog/kafka-to-pubsub-template.md:64]: Would not this be a Dataflow template?

**Comment by aaltay** [comment]
What are the next steps on this PR?

**Comment by TheNeuralBit** [comment]
Thanks for the ping Ahmet. The PR with the pipeline, https://github.com/apache/beam/pull/13112, was re-worked to make it an example rather than calling it a template

@KhaninArtur would you want to update this blog post to reflect that change?

**Comment by KhaninArtur** [comment]
@TheNeuralBit @aaltay thank you very much for bringing this blog post back to the discussion!

I will rewrite it and then ping you once again.

File: website/www/site/content/en/blog/kafka-to-pubsub-template.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+In a simple scenario, the template will create an Apache Beam pipeline that will read messages from a source Kafka server with a source topic, and stream the text messages into specified Pub/Sub destination topic. Other scenarios may need Kafka SASL/SCRAM authentication, that can be performed over plain text or SSL encrypted connection. The template supports using a single Kafka user account to authenticate in the provided source Kafka servers and topics. To support SASL authenticaton over SSL the template will need an SSL certificate location and access to a secrets vault service with Kafka username and password, currently supporting HashiCorp Vault.
+
+# Where can I run this template?
+
+There are two ways to execute the pipeline.
+1. Locally. This way has many options - run directly from your IntelliJ, or create `.jar` file and run
+it in the terminal, or use your favourite method.
+2. In [Google Cloud](https://cloud.google.com/). Using Google Cloud [Dataflow](https://cloud.google.com/dataflow) with `gcloud`
+command-line tool you can create a Flex Template out of this Beam template and execute it in Google Cloud Platform.
```
Comment (rosetn) [inline website/www/site/content/en/blog/kafka-to-pubsub-template.md:59]: Should we link to Flex Templates page? https://cloud.google.com/dataflow/docs/concepts/dataflow-templates

**Review by rosetn — CHANGES_REQUESTED** [review]

File: website/www/site/content/en/blog/kafka-to-pubsub-template.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+The template provides code examples to 
+implement simple yet powerful pipelines and also provides an out-of-the-box solution 
+that you can just _"plug'n'play"_. 
+
+We hope you will find this template useful for setting up data pipelines between Kafka and Pub/Sub.
+
+# Template specs
+
+Supported data formats:
+- Serializable plaintext formats, such as JSON
```
[the remark this answers is not in the document]
Comment (KhaninArtur) [reply website/www/site/content/en/blog/kafka-to-pubsub-template.md]: Changed to "plain text"

File: website/www/site/content/en/blog/kafka-to-pubsub-template.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+In this blog post we present an example template that creates a pipeline
+to read data from a single or multiple topics from
```
[the remark this answers is not in the document]
Comment (KhaninArtur) [reply website/www/site/content/en/blog/kafka-to-pubsub-template.md]: Added

File: website/www/site/content/en/blog/kafka-to-pubsub-template.md (hunk 2 above)
Comment (KhaninArtur) [reply website/www/site/content/en/blog/kafka-to-pubsub-template.md]: Added link to Flex Templates

File: website/www/site/content/en/blog/kafka-to-pubsub-template.md (hunk 1 above)
Comment (KhaninArtur) [reply website/www/site/content/en/blog/kafka-to-pubsub-template.md]: Changed this to Beam example

**Comment by KhaninArtur** [comment]
@TheNeuralBit @aaltay @rosetn bumping up this blog post PR to announce this enhancement included in the 2.27 release. I updated it to reflect all changes that have happened since the first draft.

File: website/www/site/content/en/blog/kafka-to-pubsub-example.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+-->
+
+In this blog post we present an example that creates a pipeline to read data from a single topic or
+multiple topics from [Apache Kafka](https://kafka.apache.org/) and write data into a topic
+in [Google Pub/Sub](https://cloud.google.com/pubsub). The example provides code samples to implement
+simple yet powerful pipelines and also provides an out-of-the-box solution that you can just _"
+plug'n'play"_.
+
+This end-to-end example is included
+in [Apache Beam release 2.27](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12319527&version=12349380)
```
Comment (aaltay) [inline website/www/site/content/en/blog/kafka-to-pubsub-example.md:34]: For 2.27.0 - I would suggest linking to https://beam.apache.org/blog/beam-2.27.0/ and adding a line item to that release post about this template.

**Review by aaltay — APPROVED** [review]
LGTM. We can merge it once @rosetn reviews and approves.

Thanky you!

File: website/www/site/content/en/blog/kafka-to-pubsub-example.md (hunk 5 above)
Comment (KhaninArtur) [reply website/www/site/content/en/blog/kafka-to-pubsub-example.md]: Thanks for the suggestion! Did this.

**Comment by aaltay** [comment]
@rosetn - Could you PTAL?

File: website/www/site/content/en/blog/kafka-to-pubsub-example.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+and can be downloaded [here](https://beam.apache.org/get-started/downloads/#2270-2020-12-22).
+
+We hope you will find this example useful for setting up data pipelines between Kafka and Pub/Sub.
+
+# Example specs
+
+Supported data formats:
+
+- Serializable plain text formats, such as JSON
+- [PubSubMessage](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage).
```
Comment (rosetn) [inline website/www/site/content/en/blog/kafka-to-pubsub-example.md:44]: Nit: You don't need any end punctuations for these list items. 

More information here: https://developers.google.com/style/lists#numbered,-lettered,-and-bulleted-lists

**Review by rosetn — APPROVED** [review]
Overall LGTM!

File: website/www/site/content/en/blog/kafka-to-pubsub-example.md (hunk 6 above)
Comment (KhaninArtur) [reply website/www/site/content/en/blog/kafka-to-pubsub-example.md]: Thank you! Fixed it.

**Pull request merged**